hyperledger/sawtooth-core
# Add API authorization and user updates to T&T Server

Adds an authorization route and update users route , along with corresponding middleware and database API updates. This completes the server-side requirements for [STL-627](https://jira.hyperledger.org/browse/STL-627)

To test:
```
$ cd sawtooth-core/families/track_and_trade/server
$ npm install
$ npm run init 
$ npm start
```
You should now be able to send the following requests (all request and response bodies are JSON):

### POST /users
_request:_
```
{
    "username": <string>,
    "password": <string>,
    "email": <string, must be a valid email>,
    "publicKey": <string>,
    "privateKey": <string>
}
```

_response:_
```
{
    "authorization": <string, JWT auth token>,
    "signing": <private key>
}
```


### POST /authorization
_request:_
```
{
    "username": <string>,
    "password": <string>
}
```

_response:_
```
{
    "authorization": <string, JWT auth token>,
    "signing": <private key>
}
```

### PATCH /users
Must add the JWT token to the `Authorization` header of this request.
_request_ (one or more of these keys):
```
{
    "username": <string>,
    "password": <string>,
    "email": <string, must be a valid email>,
    "privateKey": <string>
}
```

_response:_
```
{
    "username": <string>,
    "email": <string, must be a valid email>,
    "publicKey": <string>,
    "privateKey": <string>
}
```

If you're feeling adventurous, you can go to `localhost:8080` in your browser to explore the data you've created.
